beater/beater.go

```diff
@@ -381,31 +415,151 @@ func (s *serverRunner) run() error {
 		}()
 	}
 
-	var sourcemapStore *sourcemap.Store
 	if s.config.RumConfig.Enabled && s.config.RumConfig.SourceMapping.Enabled {
 		store, err := newSourcemapStore(s.beat.Info, s.config.RumConfig.SourceMapping, s.fleetConfig)
 		if err != nil {
 			return err
 		}
-		sourcemapStore = store
+		if s.sourcemapStore == nil {
+			s.sourcemapStore = store
+		} else {
+			*s.sourcemapStore = *store
+		}
+	} else {
+		// TODO: We have to remember to disable the sourcemap if it's
+		// no longer enabled
+		s.sourcemapStore = nil
+	}
+
+	if !s.running {
```

Comment:
This block contains the "one-time" configuration, which potentially should be moved. That might remove the need to have a "running" status for the server.